### DISCUSSION TITLE
presets repo not used

### DISCUSSION THREAD

**Discussion by Morriz** [renovatebot/renovate#15131] [Request Help] [answered]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Hi, after splitting config into shared preset `https://github.com/redkubes/renovate-config/blob/main/default.json` with content:
```
{
  "extends": ["config:base"],
  "packageRules": [
    {
      "automerge": true,
      "automergeType": "branch",
      "matchUpdateTypes": ["patch", "pin"]
    }
  ]
}
```
and creating `https://github.com/redkubes/otomi-api/blob/master/.github/renovate.json` with content:
```
{
  "extends": ["github>redkubes/renovate-config"]
}
```
The logs translate the found config as:
```
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "github>redkubes/renovate-config"
    ]
  }
}
```
but are not using it:
```
PR is not configured for automerge
```

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by Morriz** [comment]
I see no errors relating to the shared config not being accessible. Renovate has access to both repos (`renovate-config` and `otomi-api`).

**Comment by Morriz** [comment]
We are still using `master` as default branch. Does that matter? Should we add tag `master` to the config? Will try

**Comment by viceice** [comment]
shared presets need to be on default branch. check your repo settings and push it there.

**Reply by Morriz** [reply]
the shared preset is on the main branch there, and like I mentioned, it _is_ being picked up, but discarded (so it seems)

**Reply by viceice** [reply]
did you checked the logs? you would see errors when preset won't picked-up.

**Comment by Morriz** [comment]
I moved to `main` but we see only PRs created still. Strange is that for the mentioned `otomi-api` repo the logs mention that automerge is not enabled, but for another repo with the same config it _is_:
```
DEBUG: Branch status yellow(branch="renovate/font-awesome")
DEBUG: Branch status is "yellow" - skipping automerge(branch="renovate/font-awesome")
DEBUG: mergeStatus=no automerge(branch="renovate/font-awesome")
DEBUG: Ensuring PR(branch="renovate/font-awesome")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/font-awesome")
DEBUG: getBranchPr(renovate/font-awesome)(branch="renovate/font-awesome")
DEBUG: findPr(renovate/font-awesome, undefined, open)(branch="renovate/font-awesome")
DEBUG: findPr(renovate/font-awesome, undefined, closed)(branch="renovate/font-awesome")
DEBUG: Branch automerge is enabled
```

However, also there it only creates PRs.
I just migrated both repos to use `main` branch. (That should be picked up by your runners, right?)

**Comment by Morriz** [comment, marked as answer]
I see the branch protection rules are blocking...sorry!

**Reply by Morriz** [reply]
hmmmm...but it seems not possible to bypass the required review settings. Or is it? Hopefully we don't have to remove those requirements to enable automerge?

**Reply by viceice** [reply]
renovate can't bypass itself, the renovate accounts need to whitelisted. there are two accounts for the app. the committee and the app.

**Reply by Morriz** [reply]
Ok, this might not be clear, but we use GitHub hosted version of Renovate. How can it be whitelisted? We can't put credentials in repo files of course. Can we just create a `RENOVATE_TOKEN` secret that is accessible by the repo, and then renovate will use that as identity?

**Reply by viceice** [reply]
nope, that only works for self-hosted renovate.

i know on most branch protection rule field you can add normal GitHub user accounts and GitHub app (bot) accounts. you need to add both to the rule whitelist, as renovate uses two of them. one for git commits and one for API calls.

**Reply by viceice** [reply]
@rarkins what are the two name he had to add. i never used those in rules.

**Comment by Morriz** [comment]
Did your devs just change your running app? It suddenly tries to find the config in the wrong location:
```
DEBUG: migrated config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "github>renovatebot/spring-remediations",
      "github>redkubes/renovate-config"
    ]
  }
}
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/redkubes/renovate-config/contents/default.json"
}
DEBUG: Failed to retrieve default.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/redkubes/renovate-config/contents/default.json"
}
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/redkubes/renovate-config/contents/renovate.json"
}
DEBUG: Failed to retrieve renovate.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/redkubes/renovate-config/contents/renovate.json"
}
DEBUG: Preset fetch error
}
```
But it exists, right here: https://github.com/redkubes/renovate-config/blob/main/default.json. How come your request is not targeting the right location?

**Reply by viceice** [reply]
no, we didn't changed anything related to presets recently. we simply use the `contents` API to get file contents from default branch.

are both repos private and in same org?
the hosted app doesn't support reading private presets across organizations.

**Reply by Morriz** [reply]
aha, but it was reading it fine just before from another repo, even though it is private. But it was being requested from another private repo. Maybe that is why it was allowed? The last failure was a public repo requesting the private shared presets. If my assumption is correct, then is this documented?

**Reply by Morriz** [reply]
All our repos are in same org. But I will set it to public if needed, as it does not really give much sensitive info, but it would be quite a limitation.

**Reply by viceice** [reply]
you can only read private presets from private repo in same org. public repos can only use public presets (from any org) 

always use public presets if there aren't any sensitive data. split sensitive presets to different repos.
